src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PreferIsExactSpellingIsTrueForKnownApisAnalyzer.cs

```diff
@@ -0,0 +1,159 @@
+                                                                             isPortedFxCopRule: false,
+                                                                             isDataflowRule: false,
+                                                                             isEnabledByDefaultInFxCopAnalyzers: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, WideRule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterCompilationStartAction(ctx =>
+            {
+
+                if (!ctx.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out var dllImportType))
+                {
+                    return;
+                }
+
+                ctx.RegisterSymbolAction(x => AnalyzeExternMethod(x, dllImportType), SymbolKind.Method);
+            });
+        }
+
+        private static void AnalyzeExternMethod(SymbolAnalysisContext context, INamedTypeSymbol dllImportType)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (!methodSymbol.IsExtern)
+            {
+                return;
+            }
+
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Equals(dllImportType));
+
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var hasExactSpellingParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("ExactSpelling", StringComparison.Ordinal));
+            var hasCharSetParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("CharSet", StringComparison.Ordinal));
+
+            var methodName = methodSymbol.GetActualExternName(dllImportAttribute);
+            var isExactSpelling = hasExactSpellingParameter.Key is not null && hasExactSpellingParameter.Value.Kind != TypedConstantKind.Array && bool.TryParse(hasExactSpellingParameter.Value.Value?.ToString(), out var isExact) && isExact;
+            var isCharSetUnicode = hasCharSetParameter.Key is not null && hasCharSetParameter.Value.Kind != TypedConstantKind.Array && Enum.TryParse<CharSet>(hasCharSetParameter.Value.Value?.ToString(), out var actualCharSet) && actualCharSet == CharSet.Unicode;
+            var dllName = dllImportAttribute.ConstructorArguments.FirstOrDefault().Value?.ToString() ?? string.Empty;
+
+            if (!KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                return;
+            }
+
+            if (methods.Contains(methodName))
```

Comment:
@jeffhandley I think someone from the runtime team should review the core analyzer semantics in terms of the analyzer implementation matching the recommended API usage.

Reply:
That would be wonderful, I am self taught and have no clue whether I did something bad when writing these analyzers, I just went with the docs samples and learned from there.

Reply:
It seems missing some scenarios which i commented [here ](https://github.com/dotnet/roslyn-analyzers/pull/4130#discussion_r537929600), but i would defer to @elinor-fung for final expected scenarios